### PR TITLE
Fix matching logic for logs from namespace when lines = 0

### PR DESCRIPTION
**Pull request by bawjensen**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #5658
| License       | MIT
| Doc PR        | N/A (?)

### CODE REVIEW

File: lib/API/Log.js [hunk 1]
```diff
@@ -98,9 +98,11 @@ Log.stream = function(Client, id, raw, timestamp, exclusive, highlight) {
...
 
     bus.on('log:*', function(type, packet) {
-      if (id !== 'all'
-          && packet.process.name != id
-          && packet.process.pm_id != id)
+      var isMatchingProcess = id === 'all'
+        || packet.process.name === id
+        || packet.process.pm_id === id
+        || packet.process.namespace === id;
+      if (!isMatchingProcess)
```
Comment (bawjensen) [inline lib/API/Log.js:105]: separating out `isMatchingProcess` felt more readable to me - happy to revert back to something like this instead:

```js
        if (id !== 'all'
          && packet.process.name != id
          && packet.process.pm_id != id
          && packet.process.namespace != id)
```

File: test/e2e/logs/log-namespace.sh
Comment (bawjensen) [inline test/e2e/logs/log-namespace.sh:1]: I used the most recently added `test/e2e/logs/` case as a template, hoping to use the most modern test patterns

File: test/e2e/logs/log-namespace.sh [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+LOG_PATH_PREFIX="${SRC}/__log-namespace__"
+
+rm -rf "${LOG_PATH_PREFIX}"
+mkdir "${LOG_PATH_PREFIX}"
+
+$pm2 start echo.js --namespace e2e-test-log-namespace
+
+LOG_FILE_BASELINE="${LOG_PATH_PREFIX}/baseline-out.log"
+$pm2 logs e2e-test-log-namespace > $LOG_FILE_BASELINE & # backgrounded - will be stopped by `$pm2 delete all`
```
Comment (bawjensen) [inline test/e2e/logs/log-namespace.sh:16]: I haven't seen any e2e (or programmatic tests) for the `pm2 logs` command, in order to mimic those. So I'm shooting a bit in the dark with this approach of backgrounding the process.

**Review by hinogi — APPROVED** [review]
LGTM

**Comment by Unitech** [comment]
fixed via pm2@5.3.1
Update:

```bash
$ npm install pm2@5.3.1
$ pm2 update
```

**Pull request merged**